pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1334,12 +1326,6 @@ public void updateIndexingConfigFor(String tableName, TableType type, IndexingCo
     }
     tableConfig.setIndexingConfig(newConfigs);
     setExistingTableConfig(tableConfig);
-
-    if (type == TableType.REALTIME) {
```

Comment:
This was an optimization we had so that controllers do not need to set watches on HLC tables if they are not there in the cluster. Not sure why it has been removed, or if it has been moved elsewhere

Reply:
It is already checked in `setExistingTableConfig`. This check is redundant